token/token.go

```diff
@@ -317,7 +344,14 @@ func init() {
 		}
 	}
 	for _, keyword := range reservedBoolKeywords {
-		reservedKeywordMap[keyword] = func(value, org string, pos *Position) *Token {
+		f := func(value, org string, pos *Position) *Token {
+			return reservedKeywordToken(BoolType, value, org, pos)
+		}
+		reservedKeywordMap[keyword] = f
+		reservedEncKeywordMap[keyword] = f
```

Comment:
This may be intended to output keywords such as true/false as quoted strings. This is consistent in terms of quoting all values that should be output as boolean. However, we should concern that quoting a value that should originally be output as a boolean in the 1.2 spec will result in a breaking change.

Reply:
> we should concern that quoting a value that should originally be output as a boolean in the 1.2 spec will result in a breaking change

Hey @goccy Thank you for your feedback! Just to be sure, I believe this doesn't affect the boolean encoding.

I presume that the existing `reservedKeywordMap`  is supposed to "quote strings" (not bools) so that the results can be decoded back to the original strings.

Similarly, the new `reservedEncKeywordMap` is intended to "quote strings" (again, not bools) so that other legacy YAML parsers can decode the results without losing types and values.

Your confirmation/correction is much appreciated! Thanks.

Reply:
I see, only for string type, the quote is added. That seems to work 👍